### PR TITLE
Conneg altprofiles model

### PR DESCRIPTION
**Pull request by nicholascar**
completed v1 of the model, with an extension in an appendix and examples

rendered at https://raw.githack.com/w3c/dxwg/conneg-altprofiles-model/conneg-by-ap/index.html

### CODE REVIEW

**Review by larsgsvensson — CHANGES_REQUESTED** [review]
I really think this needs more work... As it is now, [§6.3](https://raw.githack.com/w3c/dxwg/conneg-altprofiles-model/conneg-by-ap/index.html#altr) and [Appendix A.2](https://raw.githack.com/w3c/dxwg/conneg-altprofiles-model/conneg-by-ap/index.html#altr-detail) are sort of hanging in mid-air. At least I cannot from the rest of the document read when and how to use the data model.

Considering that the model is normative ("This data model, MUST be used for describing the representations of a resource that conform to different profiles.") we also MUST give implementers clear advice when it is to be used and examples for how the suggested representations are used in concrete client/server interactions.

**Comment by nicholascar** [comment]
@larsgsvensson I agree that we need more "how to use the data model" so more pointers to it from the rest of the document. I'll wait until I can merge all the other approved PRs (hopefully a day or so only) and then I'll reconsider pointers to the model from elsewhere when all the other changes are through into gh-pages. It's just a bit to hard to see what's where until that's done.

**Comment by nicholascar** [comment]
As per the above commit comments, I have added:

* references throughout to AltP model
* HTTP implementation of AltP
* QSA implementation of AltP

File: conneg-by-ap/index.html
Comment (larsgsvensson) [inline conneg-by-ap/index.html:83]: This text can be removed since it's already in lines 75-76

File: conneg-by-ap/index.html
Comment (larsgsvensson) [inline conneg-by-ap/index.html]: Looking at the full text, I think we'd need a separate definition for "Functional Profile", too.

File: conneg-by-ap/index.html
Comment (larsgsvensson) [inline conneg-by-ap/index.html]: Suggestion: s/indicates an instance/indicates an instaonce of altp:Representation/

File: conneg-by-ap/index.html
Comment (larsgsvensson) [inline conneg-by-ap/index.html]: s/may indicates/may indicate/
Question: is it normative? Then it should be "MAY indicate"

File: conneg-by-ap/index.html
Comment (larsgsvensson) [inline conneg-by-ap/index.html]: Suggestion: Replace "understood to link ... given" with "the implicit relation between the link target and the value of the target attribute profile"

File: conneg-by-ap/index.html
Comment (larsgsvensson) [inline conneg-by-ap/index.html]: s/RFQ/RFC/

File: conneg-by-ap/index.html
Comment (larsgsvensson) [inline conneg-by-ap/index.html]: The way this paragraph is written, I seems to assume that QSA is about HTML ("QSA implementers could implement ... via HTTP body content in HTML"). Wouldn't that apply to HTTP conneg, too if the preferred media type is `text/html`, and what if I do QSA and ask for media type `application/pdf`? Since media types and profiles are orthogonal, we MUST be extremely careful, not to assume that one conneg implementation is geared towards one specific media type and another implementation toward others.

**Review by larsgsvensson — CHANGES_REQUESTED** [review]
Most proposed changes are purely editorial. My main concern is the text in [§7.2.2.1](https://raw.githack.com/w3c/dxwg/conneg-altprofiles-model/conneg-by-ap/index.html#qsa-altp) where I have the feeling that we're too focused on QSA being mainly for HTML representations.

File: conneg-by-ap/index.html
Comment (nicholascar) [reply conneg-by-ap/index.html]: Changed as requested

File: conneg-by-ap/index.html
Comment (nicholascar) [reply conneg-by-ap/index.html]: Changed as requested

File: conneg-by-ap/index.html
Comment (nicholascar) [reply conneg-by-ap/index.html]: It's sort-of normative in that this is the mapping given for this Functional Profile but the directive of MAY doesn't seem to work well here so I've changed the test to read:

> a *target attribute* of `profile` indicates the URI of an instance of prof:Profile, if present

File: conneg-by-ap/index.html
Comment (nicholascar) [reply conneg-by-ap/index.html]: Changed as requested

File: conneg-by-ap/index.html
Comment (nicholascar) [reply conneg-by-ap/index.html]: Changed as requested

File: conneg-by-ap/index.html
Comment (nicholascar) [reply conneg-by-ap/index.html:83]: I've re-written both overlapping paragraphs to improve wording and to de-duplicate.

**Comment by larsgsvensson** [comment]
No, I had also voiced [concerns](https://github.com/w3c/dxwg/pull/1038/commits/80c3232a056b041617f15ba752208c0a0e04a5e2#r322096702) that the current text ties QSA and HTML too tightly.
Apart from that, all my comments are addressed, though.

File: conneg-by-ap/index.html
Comment (nicholascar) [reply conneg-by-ap/index.html]: Addressed in a commit that both re-writes the questionable paragraph and gives examples in JSON as well as HTML to ensure multiple HTTP body response examples are present in the doc.

**Comment by nicholascar** [comment]
I separated addressing your last comment from the earlier purely editorial ones since it was more substantive.

**Review by larsgsvensson — APPROVED** [review]
OK, now it's much clearer.

**Review by rob-metalinkage — APPROVED** [review]

**Pull request merged**
